core/templates/pages/story-editor-page/navbar/story-editor-navbar.component.spec.ts

```diff
@@ -106,20 +125,393 @@ describe('Story editor navbar directive', () => {
+    fixture.detectChanges();
+
+    expect(component.warningsAreShown).toBe(false);
+  });
+
+  it('should toggle navigation options', () => {
+    component.showNavigationOptions = true;
+
+    component.toggleNavigationOptions();
+    fixture.detectChanges();
+
+    expect(component.showNavigationOptions).toBe(false);
+  });
+
+  it('should toggle edit options', () => {
+    component.showStoryEditOptions = true;
+
+    component.toggleStoryEditOptions();
+    fixture.detectChanges();
+
+    expect(component.showStoryEditOptions).toBe(false);
+  });
+
+  it('should navigate to main tab in story editor page', () => {
+    expect(component.activeTab).toBe(undefined);
+
+    component.selectMainTab();
+
+    expect(component.activeTab).toBe('Editor');
+  });
+
+  it('should navigate to preview tab in story editor page', () => {
+    expect(component.activeTab).toBe(undefined);
+
+    component.selectPreviewTab();
+
+    expect(component.activeTab).toBe('Preview');
+  });
+
+  it('should discard changes', () => {
+    story = storyObjectFactory.createFromBackendDict(storyBackendDict);
+
+    component.story = story;
+    let clearChangesSpy = spyOn(
+      undoRedoService, 'clearChanges').and.returnValue();
+
+    component.discardChanges();
+
+    expect(clearChangesSpy).toHaveBeenCalled();
+  });
+
+  describe('open a confirmation modal for saving chages ', () => {
+    it('should save story successfully on' +
+      'clicking save draft button', fakeAsync(() => {
+      story = storyObjectFactory.createFromBackendDict(storyBackendDict);
+      let mockStoryInitializedEventEmitter = new EventEmitter();
+
+      spyOn(storyEditorStateService, 'getStory').and.returnValue(story);
+      spyOnProperty(storyEditorStateService, 'onStoryInitialized')
+        .and.returnValue(mockStoryInitializedEventEmitter);
+      spyOn(editableStoryBackendApiService, 'changeStoryPublicationStatusAsync')
+        .and.returnValue(Promise.resolve());
+      const saveChangesSpy = spyOn(
+        storyEditorStateService, 'saveStory')
+        .and.callFake((commitMessage, successCallback, errorCallback) => {
+          if (commitMessage !== null) {
+            successCallback();
+          }
+          errorCallback('Expected a commit message but received none.');
+          return true;
+        });
+      const modalSpy = spyOn(ngbModal, 'open').and.callFake((dlg, opt) => {
+        setTimeout(opt.beforeDismiss);
+        return <NgbModalRef>(
+          { componentInstance: MockNgbModalRef,
+            result: Promise.resolve()
+          });
+      });
+
+      component.ngOnInit();
+      mockStoryInitializedEventEmitter.emit();
+      storyEditorStateService.setStory(story);
+      fixture.detectChanges();
+
+      component.commitMessage = 'commitMessage';
+      component.saveChanges();
+      tick();
+      fixture.detectChanges();
+
+      expect(modalSpy).toHaveBeenCalled();
+      expect(saveChangesSpy).toHaveBeenCalled();
```

Comment:
Can we also ensure that the successCallback was called? And check the commit message passed in?

Reply:
Resolved as per offline discussion.